model-optimizer/extensions/ops/tensor_iterator.py

```diff
@@ -393,10 +385,12 @@ def check_field(record, field):
             if not is_fully_defined(in_rec_end) or not is_fully_defined(in_rec_start):
                 iterations_count = dynamic_dimension_value
                 continue
-            if (in_rec_end - in_rec_start) / in_rec_stride > iterations_count:
-                iterations_count = (in_rec_end - in_rec_start) / in_rec_stride
 
-        return iterations_count
+            if iterations_count is not None and ceil((in_rec_end - in_rec_start) / in_rec_stride) != iterations_count:
+                raise Error("TensorIterator node {} have inputs with different iterations count".format(ti_node.id))
+            iterations_count = ceil((in_rec_end - in_rec_start) / in_rec_stride)
+
+        return iterations_count if iterations_count is not None else 1
```

Comment:
just return `iteration_count` because above we can initialize it as `dynamic_value`

Reply:
changed